### PR TITLE
Update puml in DG

### PR DESCRIPTION
**Pull request by JellyPenguinnn**

### CODE REVIEW

File: docs/diagrams/ArchitectureDiagram.puml [hunk 1]
```diff
@@ -30,4 +30,9 @@ Main -down[hidden]-> Commons
 Storage -up[STORAGE_COLOR].> Model
 Storage .right[STORAGE_COLOR].>File
 User ..> UI
+
+' Add backup and restore processes
+Model -[#orange]-> Storage : Backup
+Model <-[#purple]- Storage : Restore
+
```
Comment (Jaynon) [inline docs/diagrams/ArchitectureDiagram.puml:37]: looks good

File: docs/diagrams/ArchitectureDiagram.puml [hunk 2]
```diff
@@ -30,4 +30,9 @@ Main -down[hidden]-> Commons
 Storage -up[STORAGE_COLOR].> Model
 Storage .right[STORAGE_COLOR].>File
 User ..> UI
+
+' Add backup and restore processes
```
Comment (Al-ez) [inline docs/diagrams/ArchitectureDiagram.puml:34]: This line looks good

File: docs/diagrams/ArchitectureDiagram.puml [hunk 3]
```diff
@@ -30,4 +30,9 @@ Main -down[hidden]-> Commons
 Storage -up[STORAGE_COLOR].> Model
```
Comment (Al-ez) [inline docs/diagrams/ArchitectureDiagram.puml:30]: Good line break

File: docs/diagrams/ArchitectureDiagram.puml [hunk 4]
```diff
@@ -30,4 +30,9 @@ Main -down[hidden]-> Commons
...
 Storage .right[STORAGE_COLOR].>File
```
Comment (Al-ez) [inline docs/diagrams/ArchitectureDiagram.puml:31]: Nice coding here

File: docs/diagrams/ArchitectureDiagram.puml [hunk 5]
```diff
@@ -30,4 +30,9 @@ Main -down[hidden]-> Commons
...
 User ..> UI
+
+' Add backup and restore processes
+Model -[#orange]-> Storage : Backup
```
Comment (Al-ez) [inline docs/diagrams/ArchitectureDiagram.puml:35]: Yes backup is good

File: docs/diagrams/ArchitectureDiagram.puml [hunk 6]
```diff
@@ -30,4 +30,9 @@ Main -down[hidden]-> Commons
...
+Model <-[#purple]- Storage : Restore
```
Comment (Al-ez) [inline docs/diagrams/ArchitectureDiagram.puml:36]: This looks good

**Review by Al-ez — COMMENTED** [review]
LGTM

File: docs/diagrams/ArchitectureDiagram.puml (hunk 6 above)
Comment (colinac) [inline docs/diagrams/ArchitectureDiagram.puml:36]: Good use of UML

File: docs/diagrams/ArchitectureDiagram.puml (hunk 1 above)
Comment (colinac) [inline docs/diagrams/ArchitectureDiagram.puml:37]: Overall, apt update of uml diagram!

File: docs/diagrams/ArchitectureDiagram.puml (hunk 4 above)
Comment (colinac) [inline docs/diagrams/ArchitectureDiagram.puml:31]: Nice!

**Review by Jaynon — APPROVED** [review]
lgtm

**Comment by ArtillerySun** [comment]
LGTM

**Pull request merged**